client/src/pages/answer/AnswerPage.tsx

```diff
@@ -1,21 +1,103 @@
-import { Alert, Card } from 'antd'
+import { Alert, Button, Card, Col, Icon, Modal, Row } from 'antd'
 import moment from 'moment'
-import React, { useContext, useEffect } from 'react'
+import React, { useCallback, useContext, useEffect, useState } from 'react'
 import AnswerBlocker from '../../components/AnswerBlocker'
 import AnswerFileBrowser from '../../components/AnswerFileBrowser'
 import ArchiveDownload from '../../components/ArchiveDownload'
 import AsyncPlaceholder from '../../components/AsyncContainer'
 import FileImport from '../../components/FileImport'
+import useMomentReached from '../../hooks/useMomentReached'
 import {
   Answer,
   useGetAnswerQuery,
   useImportAnswerSourceMutation,
+  useResetAnswerMutation,
   useUploadAnswerSourceMutation
 } from '../../services/codefreak-api'
 import { messageService } from '../../services/message'
 import { displayName } from '../../services/user'
 import { DifferentUserContext } from '../task/TaskPage'
 
+const DangerZone: React.FC<{ answer: Pick<Answer, 'id' | 'deadline'> }> = ({
+  answer: { id, deadline }
+}) => {
+  const [resetAnswer, { loading: resetLoading }] = useResetAnswerMutation({
```

Comment:
We need to refetch the file preview here.

Reply:
Yeah, and also after uploading/importing files (see #346). I will do this in another PR, because this needs some work.